cluster/engine.go

```diff
@@ -404,7 +416,7 @@ func (e *Engine) updateSpecs() error {
 		return fmt.Errorf("cannot get resources for this engine, make sure %s is a Docker Engine, not a Swarm manager", e.Addr)
 	}
 
-	v, err := e.client.Version()
+	v, err := e.apiClient.ServerVersion()
 	e.CheckConnectionErr(err)
```

Comment:
Please update `CheckConnectionErr` to handle errors from engine-api. In engine-api, a connection failure message is the following, which is not covered in `CheckConnectionErr`.  

```
// ErrConnectionFailed is a error raised when the connection between the client and the server failed.
   var ErrConnectionFailed = errors.New("Cannot connect to the Docker daemon. Is the docker daemon running on this host?")        
```